WrightTools/data/_data.py

```diff
@@ -304,23 +304,57 @@ def bring_to_front(self, channel):
         new.insert(0, new.pop(channel_index))
         self.channel_names = new
 
-    def _at(self, idx, removed_shape, removed_axes, kept_axes, parent, name, **at):
-        idx = np.array(idx, dtype=object)
-        idx[np.array(removed_shape) == 1] = slice(None)
+    def at(self, parent, name, idx=None, kept_axes=None, **at) -> Union[object, None]:
+        """Return data of a subset of the data at specified axis value(s).
+
+        kwargs
+        ------
+        parent : WrightTools.Collection (optional)
+
+        name : string (optional)
+
+        **coordinates :
+        keys are axes, values lists of format [val, unit]
+
+        Returns
+        -------
+        out : wt.Data
+            Data that retains axes of all unspecified coordinates.
+
+        Example
+        -------
+        ```
+        from WrightTools import datasets
+        data = wt.open(datasets.wt5.v1p0p1_MoS2_TrEE_movie)  # axes w2, w1=wm, d2
+        zero_delay_data = data.at(d2=[0, "fs"])
+        ```
+
+        See Also
+        --------
+        Data.chop : also reduces dimensionality, but returns a collection of one or more
+        data objects. Kept axes can be sliced into a collection
+        """
+        if idx is None:
+            idx = np.array([slice(None)] * len(self._axes))
         for axis, point in at.items():
+            # DDK: is this stuff necessary?  axis must be str says docs...
+            #   this is some hidden way to deal with Axis indices?
```

Comment:
so, axis is a string because we use `**kwargs` (we call it `at`, but same difference)


I think there may technically be implementations of python that allow non-string kwarg keys to be passed in in certain circumstances (e.g. doing `foo(**{1:3})`) but CPython disallows it so probably not a thing we need to actually account for.

I feel like I remember passing as ints in the past, but that may have been an older implementation that passed a dict rather than as kwargs. (even py 2.7 explicitly disallows kwargs that aren't strings)